File: app/src/main/java/ua/meugen/android/pdfcomposer/model/events/PdfExportProgressEvent.java
package ua.meugen.android.pdfcomposer.model.events;

public class PdfExportProgressEvent {

    public final int completedPages;
    public final int totalPages;

    public PdfExportProgressEvent(
            final int completedPages,
            final int totalPages) {
        this.completedPages = completedPages;
        this.totalPages = totalPages;
    }

    @Override
    public boolean equals(final Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        final PdfExportProgressEvent that = (PdfExportProgressEvent) o;

        if (completedPages != that.completedPages) return false;
        return totalPages == that.totalPages;
    }

    @Override
    public int hashCode() {
        int result = completedPages;
        result = 31 * result + totalPages;
        return result;
    }
}

File: app/src/main/java/ua/meugen/android/pdfcomposer/model/events/PdfExportProgressEvent.kt
package ua.meugen.android.pdfcomposer.model.events

class PdfExportProgressEvent(
        val completedPages: Int,
        val totalPages: Int) {

    override fun equals(o: Any?): Boolean {
        if (this === o) return true
        if (o == null || javaClass != o.javaClass) return false

        val that = o as PdfExportProgressEvent?

        return if (completedPages != that!!.completedPages) false else totalPages == that.totalPages
    }

    override fun hashCode(): Int {
        var result = completedPages
        result = 31 * result + totalPages
        return result
    }
}
